### PR TITLE
audit: update to 3.1.4

### PR DESCRIPTION
**Pull request by MIvanchev**
#### Testing the changes
- I tested the changes in this PR: **NO**

### CODE REVIEW

**Comment by MIvanchev** [comment]
Fixes #50646 (hopefully).

**Comment by piekay** [comment]
https://git.alpinelinux.org/aports/plain/main/audit/musl.patch

Should fix the issue

**Comment by piekay** [comment]
I finally had time to test this myself:
With https://git.alpinelinux.org/aports/tree/main/audit/0003-all-get-rid-of-strndupa.patch I can get it to compile for musl

**Comment by MIvanchev** [comment]
@piekay This is a serious issue actually, I'm reporting it to upstream, I've changed the patch a bit to better reflect the original code. Thank you.

File: srcpkgs/audit/template [hunk 1]
```diff
@@ -15,13 +15,21 @@ license="GPL-2.0-or-later, LGPL-2.0-or-later"
...
 
 case "$XBPS_TARGET_MACHINE" in
 	*-musl) configure_args+=" --disable-zos-remote" ;;
 	*) ;;
 esac
 
+post_patch() {
+	# See https://github.com/linux-audit/audit-userspace/issues/381
+	case "$XBPS_TARGET_MACHINE" in
+		*-musl) patch -Np1 -i "${FILESDIR}/musl.patch" ;;
```
Comment (classabbyamp) [inline srcpkgs/audit/template:28]: if this patch works on glibc, don't apply it conditionally. conditional patches are icky

**Pull request merged**